src/theory/sep/theory_sep.cpp

```diff
@@ -1341,7 +1342,8 @@ Node TheorySep::mkUnion( TypeNode tn, std::vector< Node >& locs ) {
     for( unsigned i=0; i<locs.size(); i++ ){
       Node s = locs[i];
       Assert(!s.isNull());
-      s = NodeManager::currentNM()->mkNode( kind::SINGLETON, s );
+      Node op = NodeManager::currentNM()->mkConst(SingletonOp(s.getType()));
```

Comment:
We should add a mkSingleton utility to NodeManager for doing this.

Reply:
done